### PR TITLE
feat: Upgrade to Django 4.2

### PR DESCRIPTION
**Pull request by zubairshakoorarbisoft**
Issue: https://github.com/edx/upgrades/issues/134
Upgrade to Django 4.2
Django42 will work with `drf==3.14.0`. We are using only compatible versions for tests.
Removed celery other versions since `edx-platform` is already using latest celery version.

### CODE REVIEW

**Comment by awais786** [comment]
Remove celery 44 and drf old versions.

**Review by salman2013 — APPROVED** [review]

File: CHANGELOG.rst [hunk 1]
```diff
@@ -14,16 +14,25 @@ Change Log
...
 ~~~~~~~~~~
 
+
+[3.1.0] - 2023-07-21
+~~~~~~~~~~~~~~~~~~~~
+
+Added
++++++
+* Added Django42 support in CI.
+* Removed old versions of celery and drf version from ci.
```
Comment (iamsobanjaved) [inline CHANGELOG.rst:24]: Move this line to removed section.

File: CHANGELOG.rst [hunk 2]
```diff
@@ -14,16 +14,25 @@ Change Log
...
 +++++++
-
 * Removed Python 3.5 support.
 
 Chore
 +++++
 * Switch from ``edx-sphinx-theme`` to ``sphinx-book-theme`` since the former is
   deprecated
 * Upgraded celery to latest 5.x version.
+* Add Support for Django 4.2
```
Comment (iamsobanjaved) [inline CHANGELOG.rst:35]: This line already exists in Added section of changelog.

Comment (iamsobanjaved) [inline CHANGELOG.rst:35]: This line already exists in Added section of changelog.

File: CHANGELOG.rst [hunk 3]
```diff
@@ -14,16 +14,25 @@ Change Log
...
+
+Added
++++++
+* Added Django42 support in CI.
+* Removed old versions of celery and drf version from ci.
+
 Removed
 +++++++
-
 * Removed Python 3.5 support.
```
Comment (iamsobanjaved) [inline CHANGELOG.rst:28]: Kindly add new entry for the 3.0.0 release in the changelog. And move Python 3.5 support point and celery 5.x to that section as this is already released then but changelog was not updated.

**Review by iamsobanjaved — COMMENTED** [review]
Just a few changes in the changelog, otherwise all good.

File: CHANGELOG.rst (hunk 1 above)
Comment (zubairshakoorarbisoft) [reply CHANGELOG.rst:24]: ok

File: CHANGELOG.rst (hunk 2 above)
Comment (zubairshakoorarbisoft) [reply CHANGELOG.rst:35]: oky, removed from Chore

Comment (zubairshakoorarbisoft) [reply CHANGELOG.rst:35]: done

File: CHANGELOG.rst (hunk 3 above)
Comment (zubairshakoorarbisoft) [reply CHANGELOG.rst]: done

**Pull request merged**